### PR TITLE
Week-2: Milestone 3 - Add & Remove 

### PR DESCRIPTION
**Pull request by faizi2500**
## This pull request contains the add and remove functionality for To-do List App. ##

> We were asked to complete the following tasks for this PR
- [x] Remove all hardcoded items from the tasks array.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x] Implement a function for editing task descriptions.
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for the "Clear all completed" button (use filter() method).
- [x] Deleting a task should update all remaining items' indexes
- [x] All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/add-to-list.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  } else {
+    const id = taskList.length + 1;
+    const task = { description, completed, id };
+    taskList.push(task);
+    addTask.style.border = 'thin solid black';
+    localStorage.setItem('taskList', JSON.stringify(taskList));
+    displayTasks();
+  }
+  form.reset();
+};
```
Comment (Lameck1) [inline src/add-to-list.js:23]: - [OPTIONAL] It would be nice to prevent adding duplicate tasks. Don't you think?

File: src/index.js [hunk 2]
```diff
@@ -1,106 +1,104 @@
...
+let taskList = [];
 
 const getData = () => {
   if (localStorage.getItem('taskList') !== null) {
-    taskList = JSON.parse(localStorage.getItem('tasklist'));
+    taskList = JSON.parse(localStorage.getItem('taskList'));
   }
 };
 
 const displayTasks = () => {
```
Comment (Lameck1) [inline src/index.js:21]: - [OPTIONAL ] In order not to have too long functions, consider extracting event listeners out of the functions. This is necessary for code readability. Concise === readability.

**Review by Lameck1 — APPROVED** [review]
Hi, @faizi2500,

AWESOME job. Your implementation of the task at hand has been marvelously done. 
Your project is complete!  It's time to merge it :shipit: 

### Highlight
* Proffessional README ✔️ 
* Commit messages ✔️ (We can do better by making them precise. Utilise the option of adding description if maybe you want to be more elaborate with the commit message). Go through this [GUIDE](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) to get more insights.
* Using JS events ✔️
* Using localStorage (CRUD operations) ✔️

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (faizi2500) [reply src/index.js:21]: Hi Lameck, thanks for sharing your feedback. I will implement these optional changes over the weekend.

**Pull request merged**